monai/losses/dice.py

```diff
@@ -690,9 +696,19 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
             target = torch.squeeze(target, dim=1)
         target = target.long()
         ce_loss = self.cross_entropy(input, target)
+        self.loss_details["ce_loss"] = ce_loss.detach()
         total_loss: torch.Tensor = self.lambda_dice * dice_loss + self.lambda_ce * ce_loss
+        self.loss_details["total_loss"] = total_loss.detach()
+
         return total_loss
 
+    def get_loss_details(self) -> Dict[str, torch.Tensor]:
```

Comment:
I think if we have `print(my_loss.get_loss_details())` it'll cause the gpu tensor to be synced to cpu and have an impact on the speed? not sure if it's a good approach to fetch the intermediate cuda tensor results

Reply:
Hi @wyli ,

Here I just **detach** the loss tensors from graph, not moving it to CPU. Do you mean the `print()` function will potentially move the tensor to CPU? If so, it will impact the speed if you call the `print()` function?
```py
>>> import torch
>>> 
>>> a = torch.tensor(0.5, device="cuda:0")
>>> 
>>> b = a.detach()
>>> 
>>> print(b)
tensor(0.5000, device='cuda:0')
```

Thanks.

Reply:
I think it's true according to https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#avoid-unnecessary-cpu-gpu-synchronization

Reply:
Hi @wyli ,

So in that case, if users want a better perf, they just don't call the `print(my_loss.get_loss_details())`?
Maybe let me enhance the doc-string to make it clear?

Thanks.

Reply:
Users can use this `get_loss_details()` to tune the weights of hybrid loss as the feature request, if want the best perf of training, avoid printing the loss details?

Thanks.

Reply:
updated doc in the latest commit.

Thanks.

Reply:
I feel for debugging purposes the user can create a `DiceFocalLoss`, and call `DiceFocalLoss(...).dice(inp, target)`, and `DiceFocalLoss(...).focal(inp, target)` to get the internal values. Perhaps we improve `DiceCELoss` for the same?

Reply:
Hi @wyli ,

Thanks for your suggestion.
For workflow users, I think they can call `get_loss_details()` in a handler to print or debug instead of inheriting trainer and rewrite the `iteration()`?

Thanks.

Reply:
I think if we follow this idea, then for many other components, we would have to create temporary variables and add `get_*_details()` everywhere, for example in the networks, optimizers. Not sure if this is our recommended way to debug the components...

Reply:
@ericspod @rijobro , what do you guys think about the solution for this feature request? To tune the weights of hybrid losses. Maybe you have other ideas?

Thanks in advance.

Reply:
@ericspod @rijobro , if you guys have no more comments, I will update the PR to use separate `dice` and `ce` functions according to @wyli 's suggestion.

Thanks.

Reply:
perhaps it's a good opportunity to discuss the usability, we can gather more comments on this...

Reply:
Updated in the latest commit to provide the API to compute CE loss in `DiceCELoss`, so users can call `self.dice()` and `self.ce` for debug. Same as `self.focal()` as @wyli suggested.

Thanks.